### PR TITLE
CPU and memory use optimizations

### PR DESCRIPTION
**Pull request by rostislav**
Increase size of bitfield index, use all available memory in phases 3 and 4.

### CODE REVIEW

File: src/phase2.hpp [hunk 1]
```diff
@@ -147,7 +147,7 @@ Phase2Results RunPhase2(
         // * sort by pos
 
         auto sort_manager = std::make_unique<SortManager>(
-            memory_size / 2,
+            table_index == 2 ? memory_size : memory_size / 2,
```
Comment (arvidn) [inline src/phase2.hpp:150]: I think this warrants a comment

**Review by arvidn — COMMENTED** [review]
looks good to me. Did you run this through a heap profiler to ensure it doesn't exceed the limit?

File: src/phase2.hpp (hunk 1 above)
Comment (rostislav) [reply src/phase2.hpp:150]: Good idea, I will provide one

**Comment by rostislav** [comment]
> looks good to me. Did you run this through a heap profiler to ensure it doesn't exceed the limit?

I did not use a heap profiler, but I traced all calls to constructors and destructors of `SortManager` to ensure we stay within the limit.

**Review by hoffmang9 — APPROVED** [review]
Decent speed up in parallel plotting and single path plotting

**Pull request merged**